packages/server/lib/controllers/connection.controller.ts

```diff
@@ -506,6 +507,93 @@ class ConnectionController {
+
+            const body = req.body;
+
+            if (!body.connection_ids) {
+                errorManager.errRes(res, 'missing_connection_ids');
+                return;
+            }
+
+            if (!body.metadata) {
+                errorManager.errRes(res, 'missing_metadata');
+                return;
+            }
+
+            const { connection_ids: connectionIdArg, metadata } = body;
+
+            const connectionIds = Array.isArray(connectionIdArg) ? connectionIdArg : [connectionIdArg];
+
+            const ids: number[] = [];
+
+            for (const connectionId of connectionIds) {
+                const { success, response: connection } = await connectionService.getConnection(connectionId, providerConfigKey, environment.id);
+
+                if (!success || !connection || !connection.id) {
+                    const error = new NangoError('unknown_connections_bailed', { connectionId, providerConfigKey, environmentName: environment.name });
+                    errorManager.errResFromNangoErr(res, error);
+
+                    return;
+                }
+
+                ids.push(connection.id);
+            }
+
+            await connectionService.bulkReplaceMetadata(ids, metadata);
+
+            res.status(201).send(req.body);
+        } catch (err) {
+            next(err);
+        }
+    }
+    async bulkUpdateMetadata(req: Request, res: Response<any, Required<RequestLocals>>, next: NextFunction) {
+        try {
+            const environment = res.locals['environment'];
+            const providerConfigKey = (req.params['provider_config_key'] as string) || (req.get('Provider-Config-Key') as string);
+
+            const body = req.body;
+
+            if (!body.connection_ids) {
+                errorManager.errRes(res, 'missing_connection_ids');
+                return;
+            }
+
+            if (!body.metadata) {
+                errorManager.errRes(res, 'missing_metadata');
+                return;
+            }
+
+            const { connection_ids: connectionIdArg, metadata } = body;
+
+            const connectionIds = Array.isArray(connectionIdArg) ? connectionIdArg : [connectionIdArg];
+
+            const validConnections: Connection[] = [];
+
+            for (const connectionId of connectionIds) {
+                const { success, response: connection } = await connectionService.getConnection(connectionId, providerConfigKey, environment.id);
+
+                if (!success || !connection || !connection.id) {
+                    const error = new NangoError('unknown_connections_bailed', { connectionId, providerConfigKey, environmentName: environment.name });
+                    errorManager.errResFromNangoErr(res, error);
+
+                    return;
+                }
+
+                validConnections.push(connection);
+            }
+
+            await connectionService.bulkUpdateMetada(validConnections, metadata);
```

Comment:
could have been a transaction instead of recreating a new method

Reply:
Yeah, but wanted to be more explicit here though with the connection validity and.